### PR TITLE
move core.shared to core folder

### LINKED ISSUE #78: move core.shared to core folder

### PR DESCRIPTION
**Pull request by IKostric**
Fixes #78

Black was run automatically on all modified files. There were no meaningful changes.

### CODE REVIEW

File: moviebot/nlg/nlg.py [hunk 1]
```diff
@@ -747,29 +864,35 @@ def _user_options_remove_preference_CIN(self, CIN):
...
                     options[param_key] = [
-                        random.choice([
-                            f'Release year should be the "'
-                            f'{self._summarize_title_year(value)}".',
-                            f'Need a movie from the "'
-                            f'{self._summarize_title_year(value)}".'
-                        ])
+                        random.choice(
+                            [
+                                f'Release year should be the "'
```
Comment (NoB0) [inline moviebot/nlg/nlg.py:879]: Nit: f is not necessary here.

File: moviebot/nlg/nlg.py [hunk 2]
```diff
@@ -747,29 +864,35 @@ def _user_options_remove_preference_CIN(self, CIN):
...
+                                f'{self._summarize_title_year(value)}".',
+                                f'Need a movie from the "'
```
Comment (NoB0) [inline moviebot/nlg/nlg.py:881]: Nit: f is not necessary here.

File: moviebot/nlg/nlg.py [hunk 3]
```diff
@@ -747,29 +864,35 @@ def _user_options_remove_preference_CIN(self, CIN):
...
                     options[param_key] = [
-                        random.choice([
-                            f'Release year shouldn\'t be the "'
-                            f'{self._summarize_title_year(value)}".',
-                            f'Don\'t need a movie from the "'
-                            f'{self._summarize_title_year(value)}".'
-                        ])
+                        random.choice(
+                            [
+                                f"Release year shouldn't be the \""
```
Comment (NoB0) [inline moviebot/nlg/nlg.py:890]: Nit: f is not necessary here.

File: moviebot/nlg/nlg.py [hunk 4]
```diff
@@ -747,29 +864,35 @@ def _user_options_remove_preference_CIN(self, CIN):
...
+                                f'{self._summarize_title_year(value)}".',
+                                f"Don't need a movie from the \""
```
Comment (NoB0) [inline moviebot/nlg/nlg.py:892]: Nit: f is not necessary here.

**Review by NoB0 — COMMENTED** [review]
Many comments relate to docstrings or fstrings. Do we want to fix it in this PR or create subsequent issues?

**Comment by IKostric** [comment]
@NoB0 The idea was to address only the issue and nothing else. I ran black to autmatically format affected files without making any other changes. I figured it might be easier to see in the future PRs what manual changes we actually made, without black interference. 
However, I will address the review comments you made so we don't have to do it again. ;)

File: moviebot/agent/agent.py [hunk 5]
```diff
@@ -181,50 +203,65 @@ def continue_dialogue(self, user_utterance, user_options, user_fname=None):
...
             print(
-                str(self.dialogue_manager.dialogue_state_tracker.dialogue_state)
+                str(
+                    self.dialogue_manager.dialogue_state_tracker.dialogue_state
+                )
             )
             print(
-                str(self.dialogue_manager.dialogue_state_tracker.
-                    dialogue_context))
+                str(
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/agent/agent.py:243]: Outdated

File: moviebot/agent/agent.py [hunk 6]
```diff
@@ -181,50 +203,65 @@ def continue_dialogue(self, user_utterance, user_options, user_fname=None):
...
+            agent_dacts, dialogue_state=dialogue_state, user_fname=user_fname
+        )
         self.dialogue_manager.dialogue_state_tracker.dialogue_context.add_utterance(
-            AgentUtterance({'text': agent_response}))
+            AgentUtterance({"text": agent_response})
+        )
         if not self.isBot:
             print(
-                str(self.dialogue_manager.dialogue_state_tracker.dialogue_state)
+                str(
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/agent/agent.py:238]: Outdated

File: moviebot/agent/agent.py [hunk 7]
```diff
@@ -181,50 +203,65 @@ def continue_dialogue(self, user_utterance, user_options, user_fname=None):
 
         Args:
             user_utterance: The input received from the user
-            user_options: 
+            user_options:
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/agent/agent.py:206]: Created Issue

File: moviebot/agent/agent.py [hunk 8]
```diff
@@ -1,17 +1,17 @@
 """Types of conversational agents are available here."""
 
-__author__ = 'Javeria Habib'
+__author__ = "Javeria Habib"
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/agent/agent.py:3]: Outdated

File: moviebot/dialogue_manager/dialogue_act.py [hunk 9]
```diff
@@ -1,10 +1,10 @@
 """Dialogue Act defines the action user or agent takes during the conversation.
 """
 
-__author__ = 'Javeria Habib'
+__author__ = "Javeria Habib"
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/dialogue_manager/dialogue_act.py:4]: Outdated

File: moviebot/dialogue_manager/dialogue_act.py [hunk 10]
```diff
@@ -36,8 +37,11 @@ def __str__(self):
 
         """
         if self.intent:
-            return str(self.intent) + \
-                   '(' + \
-                   ', '.join([str(param) for param in self.params]) + ')'
+            return (
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/dialogue_manager/dialogue_act.py:40]: Done

File: moviebot/dialogue_manager/dialogue_manager.py [hunk 11]
```diff
@@ -5,10 +5,12 @@
 
 __author__ = "Javeria Habib"
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/dialogue_manager/dialogue_manager.py:6]: Outdated

File: moviebot/dialogue_manager/dialogue_policy.py [hunk 12]
```diff
@@ -199,27 +253,30 @@ def next_action(self, dialogue_state, dialogue_context=None, restart=False):
...
+                    )
         return agent_dacts
 
     def _generate_examples(self, database_result, slot):
         """
 
         Args:
-            database_result: 
-            slot: 
+            database_result:
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/dialogue_manager/dialogue_policy.py:270]: Separate Issue

File: moviebot/dialogue_manager/dialogue_policy.py [hunk 13]
```diff
@@ -199,27 +253,30 @@ def next_action(self, dialogue_state, dialogue_context=None, restart=False):
...
+            slot:
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/dialogue_manager/dialogue_policy.py:271]: Separate issue

File: moviebot/dialogue_manager/dialogue_state_tracker.py [hunk 14]
```diff
@@ -52,23 +53,28 @@ def _add_year_CIN(self, param):
         """
         if param.value in Values.__dict__.values():
             return param.value
-        value = str(param.op) + ' ' + param.value
+        value = str(param.op) + " " + param.value
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/dialogue_manager/dialogue_state_tracker.py:56]: Done

File: moviebot/dialogue_manager/dialogue_state_tracker.py [hunk 15]
```diff
@@ -1,17 +1,17 @@
 """Dialogue State Tracker updates the current dialogue state."""
 
-__author__ = 'Javeria Habib'
+__author__ = "Javeria Habib"
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/dialogue_manager/dialogue_state_tracker.py:3]: Outdated

File: moviebot/nlg/nlg.py [hunk 16]
```diff
@@ -521,28 +604,33 @@ def _user_options_remove_preference(self, dual_params):
...
-                            random.choice([
-                                f'I want {_a_an} "{value}" '
-                                f'genre movie.',
-                                f'I would prefer {_a_an} "{value}" '
-                                f'genre film.'
-                            ])
+                            random.choice(
+                                [
+                                    f'I want {_a_an} "{value}" '
+                                    f"genre movie.",
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlg/nlg.py:618]: Done

File: moviebot/nlg/nlg.py [hunk 17]
```diff
@@ -521,28 +604,33 @@ def _user_options_remove_preference(self, dual_params):
...
                         options[param_key] = [
-                            random.choice([
-                                f'Don\'t want {_a_an} "{value}" genre'
-                                f'movie.', f'Won\'t prefer {_a_an} "{value}" '
-                                f'genre film.'
-                            ])
+                            random.choice(
+                                [
+                                    f'Don\'t want {_a_an} "{value}" genre'
+                                    f"movie.",
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlg/nlg.py:629]: Done

File: moviebot/nlg/nlg.py [hunk 18]
```diff
@@ -560,26 +648,32 @@ def _user_options_remove_preference(self, dual_params):
...
                     else:
                         options[param_key] = [
-                            random.choice([
-                                f'Don\'t need movies based on '
-                                f'"{value}".',
-                                f'No need of {_a_an} "{value}" film.'
-                            ])
+                            random.choice(
+                                [
+                                    f"Don't need movies based on "
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlg/nlg.py:653]: Done

File: moviebot/nlg/nlg.py [hunk 19]
```diff
@@ -588,73 +682,83 @@ def _user_options_remove_preference(self, dual_params):
...
                         options[param_key] = [
-                            random.choice([
-                                f'Release year should be the "'
-                                f'{self._summarize_title_year(value)}".',
-                                f'Need a movie from the "'
-                                f'{self._summarize_title_year(value)}".'
-                            ])
+                            random.choice(
+                                [
+                                    f'Release year should be the "'
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlg/nlg.py:697]: Done

File: moviebot/nlg/nlg.py [hunk 20]
```diff
@@ -588,73 +682,83 @@ def _user_options_remove_preference(self, dual_params):
...
+                                    f'{self._summarize_title_year(value)}".',
+                                    f'Need a movie from the "'
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlg/nlg.py:699]: Done

File: moviebot/nlg/nlg.py [hunk 21]
```diff
@@ -588,73 +682,83 @@ def _user_options_remove_preference(self, dual_params):
...
-                                f'Release year shouldn\'t be the "'
-                                f'{self._summarize_title_year(value)}".',
-                                f'Don\'t need a movie from the "'
-                                f'{self._summarize_title_year(value)}".'
-                            ])
+                            random.choice(
+                                [
+                                    f"Release year shouldn't be the \""
+                                    f'{self._summarize_title_year(value)}".',
+                                    f"Don't need a movie from the \""
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlg/nlg.py:710]: Done

File: moviebot/nlg/nlg.py (hunk 1 above)
Comment (IKostric) [reply moviebot/nlg/nlg.py:879]: Done

File: moviebot/nlg/nlg.py (hunk 2 above)
Comment (IKostric) [reply moviebot/nlg/nlg.py:881]: Done

File: moviebot/nlg/nlg.py (hunk 3 above)
Comment (IKostric) [reply moviebot/nlg/nlg.py:890]: Done

File: moviebot/nlg/nlg.py (hunk 4 above)
Comment (IKostric) [reply moviebot/nlg/nlg.py:892]: Done

File: moviebot/nlg/nlg.py [hunk 22]
```diff
@@ -394,71 +434,108 @@ def _clarify_CIN(self, CIN, agent_dact):
...
+                    response + " starring " + CIN[Slots.ACTORS.value].title()
+                )
         return response.strip()
 
     def _user_options_continue(self, agent_dact):
         """Give user options to continue when needed
 
         Args:
-            agent_dact: 
+            agent_dact:
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlg/nlg.py:538]: Separate issue

File: moviebot/nlg/nlg.py [hunk 23]
```diff
@@ -271,111 +291,131 @@ def generate_output(self,
...
         else:
-            return f'{value} minutes'
+            return f"{value} minutes"
 
     def _summarize_title_year(self, value):
         """
 
         Args:
-            value: 
+            value:
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlg/nlg.py:405]: separate issue

File: moviebot/nlg/nlg.py [hunk 24]
```diff
@@ -271,111 +291,131 @@ def generate_output(self,
...
             return " ".join([str(dact) for dact in agent_dacts]), user_options
-        return '\n\n'.join(utterance), user_options
+        return "\n\n".join(utterance), user_options
 
     def _summarize_duration(self, value):
         """
 
         Args:
-            value: 
+            value:
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlg/nlg.py:384]: sepparate issue

File: moviebot/nlu/nlu.py [hunk 25]
```diff
@@ -2,16 +2,16 @@
 NLU understands the user requirements and intents for the system to generate an
 appropriate response."""
 
-__author__ = 'Javeria Habib'
+__author__ = "Javeria Habib"
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlu/nlu.py:5]: outdated

File: moviebot/nlu/user_intents_checker.py [hunk 26]
```diff
@@ -1,6 +1,6 @@
 """ This file contains the main functions for checking the user intents."""
 
-__author__ = 'Javeria Habib'
+__author__ = "Javeria Habib"
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlu/user_intents_checker.py:3]: outdated

File: moviebot/nlg/nlg.py [hunk 27]
```diff
@@ -271,111 +291,131 @@ def generate_output(self,
...
-                            f'"{param.value}"?'
-                        ])
-                    elif param.slot == 'deny':
-                        intent_response = random.choice([
-                            f'Would you want to know more about '
-                            f'"{param.value}"?'
-                        ])
+                        intent_response = random.choice(
+                            [
+                                f"What would you like to know "
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlg/nlg.py:325]: Done

File: moviebot/nlg/nlg.py [hunk 28]
```diff
@@ -271,111 +291,131 @@ def generate_output(self,
...
-                    f'{"other " if dialogue_state.items_in_context else ""}'
-                    f'{self._clarify_CIN(CIN, agent_dact)}.',
-                    f'Sorry, I couldn\'t find any '
-                    f'{"other " if dialogue_state.items_in_context else ""}'
-                    f'{self._clarify_CIN(CIN, agent_dact)}.'
-                ])
-                intent_response += ' Please select from the list of options to continue.'
+                intent_response = random.choice(
+                    [
+                        f"Sorry, I don't have any "
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlg/nlg.py:298]: Done

File: moviebot/nlg/nlg.py [hunk 29]
```diff
@@ -271,111 +291,131 @@ def generate_output(self,
...
+                        f'{"other " if dialogue_state.items_in_context else ""}'
+                        f"{self._clarify_CIN(CIN, agent_dact)}.",
+                        f"Sorry, I couldn't find any "
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlg/nlg.py:301]: Done

File: moviebot/nlg/nlg.py [hunk 30]
```diff
@@ -271,111 +291,131 @@ def generate_output(self,
...
+                            [
+                                f"What would you like to know "
+                                f"about "
+                                f'"{param.value}"?'
+                            ]
+                        )
+                    elif param.slot == "deny":
+                        intent_response = random.choice(
+                            [
+                                f"Would you want to know more about "
```
[the remark this answers is not in the document]
Comment (IKostric) [reply moviebot/nlg/nlg.py:333]: Done

**Comment by IKostric** [comment]
Fixed f-strings.
Created separate issues for method signature types and docstring fixes.

**Review by WerLaj — APPROVED** [review]
LGTM

**Review by NoB0 — APPROVED** [review]
LGTM.

**Pull request merged**